src/libs/TransactionPreviewUtils.ts

```diff
@@ -0,0 +1,326 @@
+    abandonReviewDuplicateTransactions();
+    const comparisonResult = compareDuplicateTransactionFields(reviewingTransactionID, transaction?.reportID, transaction?.transactionID ?? reviewingTransactionID);
+    setReviewDuplicatesKey({...comparisonResult.keep, duplicates, transactionID: transaction?.transactionID, reportID: transaction?.reportID});
+
+    if (comparisonResult.change.merchant) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_MERCHANT_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.category) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_CATEGORY_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.tag) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_TAG_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.description) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_DESCRIPTION_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.taxCode) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_TAX_CODE_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.billable) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_BILLABLE_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+    if (comparisonResult.change.reimbursable) {
+        return ROUTES.TRANSACTION_DUPLICATE_REVIEW_REIMBURSABLE_PAGE.getRoute(route.params?.threadReportID, backTo);
+    }
+
+    return ROUTES.TRANSACTION_DUPLICATE_CONFIRMATION_PAGE.getRoute(route.params?.threadReportID, backTo);
+};
+
+type TranslationPathOrText = {
+    translationPath?: TranslationPaths;
+    text?: string;
+};
+
+const dotSeparator: TranslationPathOrText = {text: ` ${CONST.DOT_SEPARATOR} `};
+
+function getTransactionPreviewTextAndTranslationPaths({
+    iouReport,
+    transaction,
+    action,
+    violations,
+    transactions,
+    isBillSplit,
+    shouldShowRBR,
+    violationMessage,
+}: {
+    iouReport: OnyxEntry<OnyxTypes.Report>;
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
+    action: OnyxTypes.ReportAction;
+    violations: OnyxTypes.TransactionViolations;
+    transactions: Partial<TransactionDetails>;
+    isBillSplit: boolean;
+    shouldShowRBR: boolean;
+    violationMessage?: string;
+}) {
+    const isFetchingWaypoints = isFetchingWaypointsFromServer(transaction);
+    const isTransactionOnHold = isOnHold(transaction);
+    const isTransactionMadeWithCard = isCardTransaction(transaction);
+    const isMoneyRequestSettled = isSettled(iouReport?.reportID);
+    const isSettlementOrApprovalPartial = !!iouReport?.pendingFields?.partial;
+    const isPartialHold = isSettlementOrApprovalPartial && isTransactionOnHold;
+    // We don't use isOnHold because it's true for duplicated transaction too and we only want to show hold message if the transaction is truly on hold
```

Comment:
```suggestion

    // We don't use isOnHold because it's true for duplicated transaction too and we only want to show hold message if the transaction is truly on hold
```